reana_server/reana_admin/check_workflows.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2022 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA Server administrative tool - check workflows command."""
+
+
+import datetime
+import logging
+import sys
+from typing import Optional, List, Dict, Tuple
+from functools import partial
+
+import click
+from reana_commons.k8s.api_client import current_k8s_corev1_api_client
+from reana_commons.config import REANA_RUNTIME_KUBERNETES_NAMESPACE
+from reana_db.database import Session
+from reana_db.models import (
+    Workflow,
+    RunStatus,
+)
+from kubernetes.client import V1Pod
+from kubernetes.client.rest import ApiException
+from sqlalchemy.orm import Query
+
+from reana_server.reana_admin.consumer import CollectingConsumer
+
+CheckResult = Tuple[bool, str]
+WorkflowCheckResult = Tuple[Workflow, List[CheckResult]]
+
+
+def _display(validation_results: List[WorkflowCheckResult]):
+    for workflow_result in validation_results:
+        workflow = workflow_result[0]
+        click.secho(f"- {str(workflow.id_)}\n  DB status: {workflow.status}")
+
+        click.secho("  Check results:\n")
+        check_results = workflow_result[1]
+        for check_result in check_results:
+            sign = "+" if check_result[0] else "-"
+            color = "green" if check_result[0] else "red"
+            reason = check_result[1]
+            click.secho(f"   {sign} {reason}\n", fg=color)
+
+
+def _get_all_pods() -> List[V1Pod]:
+    try:
+        response = current_k8s_corev1_api_client.list_namespaced_pod(
+            namespace=REANA_RUNTIME_KUBERNETES_NAMESPACE
+        )
+        return response.items
+    except ApiException as error:
+        click.secho(
+            "Couldn't fetch list of pods due to an error.",
+            fg="red",
+        )
+        logging.exception(error)
+        return []
+
+
+def _collect_messages_from_scheduler_queue(filtered_workflows: Query) -> Dict:
+    scheduler_messages_collector = CollectingConsumer(
+        queue_name="workflow-submission",
+        key="workflow_id_or_name",
+        values_to_collect=[str(workflow.id_) for workflow in filtered_workflows],
+    )
+    scheduler_messages_collector.run()
+    return scheduler_messages_collector.messages
+
+
+def _message_is_in_scheduler_queue(workflow, pods, scheduler_messages) -> CheckResult:
+    if workflow.id_ not in scheduler_messages:
+        return False, "Message is not found in workflow-submission queue."
+    else:
+        return True, "Message is found in workflow-submission queue."
+
+
+def _pods_dont_exist(workflow, pods, scheduler_messages) -> CheckResult:
```

Comment:
This comment is related to my comment below regarding `in_sync_workflows`. I think we could return only the errors in these `check` functions, since as I mentioned below successful output may not be that relevant. Also it would increase the readability of these functions, because now after a quick look into these functions it's not clear which return message is the error and which is not

Reply:
I replace booleans with exceptions. I think this will be more intuitive for whoever reads a code. What do you think?

Reply:
Looks much better indeed!